src/test/java/org/mule/extension/api/introspection/declaration/ComplexExtensionDeclarationTestCase.java

```diff
@@ -0,0 +1,127 @@
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.sameInstance;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.hasSize;
+import static org.junit.Assert.assertThat;
+import static org.mule.extension.api.introspection.declaration.tck.TestHttpConnectorDeclarer.EXTENSION_DESCRIPTION;
+import static org.mule.extension.api.introspection.declaration.tck.TestHttpConnectorDeclarer.EXTENSION_NAME;
+import static org.mule.extension.api.introspection.declaration.tck.TestHttpConnectorDeclarer.LISTENER_CONFIG_DESCRIPTION;
+import static org.mule.extension.api.introspection.declaration.tck.TestHttpConnectorDeclarer.LISTENER_CONFIG_NAME;
+import static org.mule.extension.api.introspection.declaration.tck.TestHttpConnectorDeclarer.LISTEN_MESSAGE_SOURCE;
+import static org.mule.extension.api.introspection.declaration.tck.TestHttpConnectorDeclarer.PATH;
+import static org.mule.extension.api.introspection.declaration.tck.TestHttpConnectorDeclarer.PORT;
+import static org.mule.extension.api.introspection.declaration.tck.TestHttpConnectorDeclarer.REQUESTER_CONFIG_DESCRIPTION;
+import static org.mule.extension.api.introspection.declaration.tck.TestHttpConnectorDeclarer.REQUESTER_CONFIG_NAME;
+import static org.mule.extension.api.introspection.declaration.tck.TestHttpConnectorDeclarer.REQUESTER_CONNECTION_PROVIDER_CONFIG_TYPE;
+import static org.mule.extension.api.introspection.declaration.tck.TestHttpConnectorDeclarer.REQUESTER_CONNECTION_PROVIDER_CONNECTION_TYPE;
+import static org.mule.extension.api.introspection.declaration.tck.TestHttpConnectorDeclarer.REQUESTER_PROVIDER;
+import static org.mule.extension.api.introspection.declaration.tck.TestHttpConnectorDeclarer.REQUEST_OPERATION_NAME;
+import static org.mule.extension.api.introspection.declaration.tck.TestHttpConnectorDeclarer.VENDOR;
+import static org.mule.extension.api.introspection.declaration.tck.TestHttpConnectorDeclarer.VERSION;
+import org.mule.extension.api.introspection.declaration.fluent.ConfigurationDeclaration;
+import org.mule.extension.api.introspection.declaration.fluent.ConnectionProviderDeclaration;
+import org.mule.extension.api.introspection.declaration.fluent.ExtensionDeclaration;
+import org.mule.extension.api.introspection.declaration.fluent.OperationDeclaration;
+import org.mule.extension.api.introspection.declaration.fluent.ParameterDeclaration;
+import org.mule.extension.api.introspection.declaration.fluent.SourceDeclaration;
+import org.mule.extension.api.introspection.declaration.tck.TestHttpConnectorDeclarer;
+import org.mule.metadata.api.model.BinaryType;
+import org.mule.metadata.api.model.NumberType;
+import org.mule.metadata.api.model.ObjectType;
+import org.mule.metadata.api.model.StringType;
+
+import java.io.Serializable;
+
+import org.junit.Test;
+import org.omg.CORBA.portable.InputStream;
+
+public class ComplexExtensionDeclarationTestCase extends BaseDeclarationTestCase
```

Comment:
Does it make sense to have a mixed test case where the are global operations/sources/providers and specific ones in certain configs? I think that would be the case for something like DB.